consensus/state.go

```diff
@@ -32,14 +33,18 @@ const (
 var hasherPool = &sync.Pool{New: func() interface{} { return types.NewHasher() }}
 
 // A Store stores blocks, siacoin outputs, siafund outputs, and file contracts.
+//
+// Store methods do not return errors. If a store encounters an error, it should
+// save the error and thereafter return empty values from all methods. It is the
+// caller's responsibility to check this error, and, if non-nil, discard any
+// results computed.
 type Store interface {
```

Comment:
Should these and the functions that use them, like `ApplyState`, return errors instead? The additional context may help when tracking down potential bugs.

Reply:
whoops, merged before I refreshed and saw comments.

`Store` originally had an `Err` method for retrieving the stored error, but you don't actually need that to be part of the consensus interface, because consensus doesn't care about it. Instead, we handle that in the concrete type. That's what `WithConsensus` does -- it gives you an implementation of `consensus.Store` with a sticky error, and returns it at the end.